Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java

```diff
@@ -327,9 +333,16 @@ void getSupportedPaymentMethods(final FragmentActivity activity, final GetSuppor
     void getSupportedCardTypes(final GetSupportedCardTypesCallback callback) {
         braintreeClient.getConfiguration((configuration, error) -> {
             if (configuration != null) {
-                List<String> supportedCardTypes = new ArrayList<>(configuration.getSupportedCardTypes());
+                List<CardType> supportedCardTypes = new ArrayList<>();
+                for (String cardTypeAsString : configuration.getSupportedCardTypes()) {
+                    CardType cardType = paymentMethodInspector.parseCardType(cardTypeAsString);
+                    if (cardType != null) {
+                        supportedCardTypes.add(cardType);
+                    }
+                }
+
                 if (!configuration.isUnionPayEnabled()) {
-                    supportedCardTypes.remove(DropInPaymentMethodType.UNIONPAY.getCanonicalName());
+                    supportedCardTypes.remove(CardType.UNIONPAY);
```

Comment:
If union pay is not enabled, and we try to remove this value from the list, but it doesn't exist in the list - wouldn't we get a crash here?

Reply:
Java `List` will actually no-op in this case. It would throw though if we were removing by index and the index was out of bounds.

Reply:
Awesome - thanks for confirming